### ISSUE TITLE
Add Syntakt support

### ISSUE THREAD

**Issue by dagargo** [dagargo/elektroid#60]

**Comment by dagargo** [comment]
Added in a75b02bd6c166a7d37de916408dfb59239bc6049.

Could anyone check if this is working?

**Comment by dagargo** [comment]
Changed to be in the JSON devices file. See 5adef1daaea0890840adc7b26b4d713e9abb7ae2.

**Comment by szszoke** [comment]
I will give it a try.

**Comment by szszoke** [comment]
This is what I get if I press the refresh button (in the second row) after I select Syntakt in the dropdown:

```
Thread 1 "elektroid" received signal SIGSEGV, Segmentation fault.
elektroid_remote_check_selection (data=<optimized out>) at elektroid.c:1328
1328	  gboolean move_impl = remote_browser.fs_ops->move ? TRUE : FALSE;
```

**Comment by dagargo** [comment]
Solved in 2ee464534f7070f0910326eb5a4be201e46cc112.

Could you try again?

**Comment by szszoke** [comment]
The original error is fixed, there is another one that can be reproduced the same way as the first one:

```
Thread 1 "elektroid" received signal SIGSEGV, Segmentation fault.
0x000055555555fae2 in browser_load_dir (data=0x55555557a1c0 <remote_browser>) at browser.c:280
280	  if (browser->fs_ops->readdir (&iter, browser->dir, browser->data))
```

**Comment by dagargo** [comment]
It'd be much better if you could try to solve it yourself. Let me explain how.

Every device implements different filesystems and storage options. Historically, these were part of the C code but now device definition are now stored in a JSON file. Take a look at the end of the `README.md` for the syntax and the meaning of the different fields.

The main advantages of having this in a text file is that you don't need to compile Elektroid every time. Just modify the file and restart Elektroid. In my system, the file is located at `/usr/local/share/elektroid/res/devices.json` but it might be different in your case.

In case of doubt, do not hesitate to ask.

**Comment by dagargo** [comment]
But the JSON file looks right to me.

**Comment by szszoke** [comment]
At this point we then assume that the SysEx format for the Syntakt is compatible with the other Digis - correct?

**Comment by szszoke** [comment]
Your last change fixed the issue. I made a mistake while testing.

I just comliled the project and run the executable from `src` so the JSON file on my system was never updated.

It works if I install the latest build.

**Comment by dagargo** [comment]
I've been told it is fully compatible. In fact, all modern Elektron devices are supported and have been reported to work.

Is the filesystem combo populated?

Can you try the different filesystems independently?

**Comment by szszoke** [comment]
I tried the following:
- list, download and upload projects
- list and browse sound folders
- download and upload sounds
- play everything that I uploaded back on the Syntakt

All of the above worked.

There are two things though that I noticed:
- right click and `Play` on a sound or patch in the left browser does nothing
- uploaded items have a `Completed` status but the progress is 0%

![image](https://user-images.githubusercontent.com/2766129/170688276-86a7c482-9341-4c86-9dfe-b532cdf99ff5.png)

**Comment by dagargo** [comment]
First and foremost, thank you so so much for showing your support to Elektroid and Overwitch. :pray:  :heart: 

We're almost there!

> right click and Play on a sound or patch in the left browser does nothing

Fixed. Play menu entry is disabled now unless there is only one sample selected.

> uploaded items have a Completed status but the progress is 0%

I'm not sure what's going on here perhaps but I think I might have found the culprit. Could you post the output of this command, please?

```
$ unzip -c '909 BD.stsnd' manifest.json
```

This will cat the content of the file `manifest.json` inside `909 BD.stsnd`. It should look like this.

```
$ unzip -c 'SNARE JAY.dtsnd' manifest.json
Archive:  SNARE JAY.dtsnd
  inflating: manifest.json           
{
  "FormatVersion" : "1.0",
  "ProductType" : [
    "12"
  ],
  "Payload" : "SNARE JAY",
  "FileType" : "Sound",
  "FirmwareVersion" : "1.30B"
}
```

**Comment by dagargo** [comment]
> uploaded items have a Completed status but the progress is 0%

Luckily, fixed in 87d710a1414f228afb2cf9b27c4c9e78fd742097.

Let me know if it works for you.

**Comment by szszoke** [comment]
It did. Both project and sound uploads show a 100% progress once the tasks are completed.

**Comment by dagargo** [comment]
Great! I'm closing this as solved and I'll release version 2.1 during the following weekend.

Let me know if you have further issues.

And thanks for checking this out! :heart:
